### PR TITLE
fix(fossid-webapp): Accept path excludes for files with a '-'

### PR DESCRIPTION
**Pull request by nnobelis**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/fossid/Utils.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ import org.ossreviewtoolkit.model.config.PathExclude
 
 private val DIRECTORY_REGEX = "(?<directory>(?:\\w+/)*\\w+)/?(?<starstar>\\*\\*)?".toRegex()
 private val EXTENSION_REGEX = "\\*\\.(?<extension>\\w+)".toRegex()
-private val FILE_REGEX = "(?<file>[\\w.]+)".toRegex()
+private val FILE_REGEX = "(?<file>[\\w.\\-]+)".toRegex()
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/Utils.kt:31]: Could this be simplified to just not allow slashed, in which case the path exclude pattern should always point to a file? Like `"(?<file>[^/]+)".toRegex()`.

Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/Utils.kt:31]: I'm specifically asking was there are more non-word characters that are valid in file names, so maintaining this as an "allow-list" becomes cumbersome.

Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/Utils.kt]: Yeah I thought about it. For instance if somebody has Unicode characters...
Fixed.

File: scanner/src/test/kotlin/scanners/fossid/MapIgnoreRulesTest.kt [hunk 2]
```diff
@@ -140,6 +140,21 @@ class MapIgnoreRulesTest : WordSpec({
             issues should beEmpty()
         }
 
+        "map rule with files (with '-' in their names" {
```
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/fossid/MapIgnoreRulesTest.kt]: The test name lacks a closing `)`.

File: scanner/src/main/kotlin/scanners/fossid/Utils.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ import org.ossreviewtoolkit.model.config.PathExclude
 
 private val DIRECTORY_REGEX = "(?<directory>(?:\\w+/)*\\w+)/?(?<starstar>\\*\\*)?".toRegex()
 private val EXTENSION_REGEX = "\\*\\.(?<extension>\\w+)".toRegex()
-private val FILE_REGEX = "(?<file>[\\w.]+)".toRegex()
+private val FILE_REGEX = "(?<file>[^/]+)".toRegex()
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/Utils.kt:31]: Please add to the commit message body: "Generally allow all characters except for a slash for file names now."

File: scanner/src/test/kotlin/scanners/fossid/MapIgnoreRulesTest.kt [hunk 4]
```diff
@@ -125,7 +125,7 @@ class MapIgnoreRulesTest : WordSpec({
             issues should beEmpty()
         }
 
-        "map rule with files (with '.' in the names" {
+        "map rule with files (with '.' in their names" {
```
Comment (sschuberth) [inline scanner/src/test/kotlin/scanners/fossid/MapIgnoreRulesTest.kt:128]: While at it, please also add the missing closing `)` here.

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Review by sschuberth — APPROVED** [review]

File: scanner/src/main/kotlin/scanners/fossid/Utils.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/Utils.kt]: BTW, you should probably do something similar for `EXTENSION_REGEX`. Also, for a path exclude pattern like "*.tar.gz", what is `EXTENSION_REGEX` supposed to match / return? Really "tar.gz", or maybe just "gz"?

**Pull request merged**